src/test/java/org/apache/commons/codec/language/bm/PhoneticEngineTest.java

```diff
@@ -70,4 +80,13 @@ public void testEncode(final String name, final String phoneticExpected, final N
             }
         }
     }
+
+    @ParameterizedTest
+    @MethodSource("invalidData")
+    public void testInvalidEncode(final String input, final NameType nameType, final RuleType ruleType,
+                                  final boolean concat, final int maxPhonemes) {
+        final PhoneticEngine engine = new PhoneticEngine(nameType, ruleType, concat, maxPhonemes);
+
+        assertDoesNotThrow(() -> engine.encode(input));
```

Comment:
Why is `assertDoesNotThrow` used here? The test method does nothing else after the call to encode so there is nothing else to run. Or, am I missing something?

Reply:
Originally I just want to test if the invalid input still throw any unexpected exceptions. But yes, I do agree the result also needed to be checked. I have changed the test case a little bit, as well as for other PRs. Thanks for your suggestions.